project/src/main/world/creature/creature.gd

```diff
@@ -119,6 +119,8 @@ func _ready() -> void:
 	add_child(_creature_outline)
 	
 	creature_visuals = $CreatureOutline.creature_visuals
+	$ChatIconHook.creature_visuals_path = $ChatIconHook.get_path_to(creature_visuals)
+	$CollisionShape2D.creature_visuals_path = $CollisionShape2D.get_path_to(creature_visuals)
```

Comment:
Use the `_collision_shape` field, and consider creating a new `chat_icon_hook` field.

Reply:
Rewrote logic to use `_collision_shape`, `_chat_icon_hook` fields.